### ISSUE TITLE
API-MS-WIN-CORE-WINRT-STRING-L1-1-0.DLL missing

### ISSUE THREAD

**Issue by Nassiel** [dotnet/coreclr#157]
Hi again,

After compiling everything, I tried to execute test but it fails. When I try to execute directly CoreConsole.exe I receive an error about this Api dll of the Kernel:
The program can't start because
API-MS-WIN-CORE-WINRT-STRING-L1-1-0.DLL is missing from your computer.
Try reinstalling the program to fix the problem.

Of course, reinstalling is not the solution. I think that he's calling to WINRT api, which doesn't exists in Windows 7 (my SO) so if I'm not wrong, this file only works on Windows 8 or upper.

The question is, are there any option to compile for windows 7 kernel api?

Thanks!

PD: I installed all redist packages x86 and x64 from 2005 to 2013.

**Comment by akoeplinger** [comment]
See https://github.com/dotnet/coreclr/issues/65

**Comment by Nassiel** [comment]
.

**Comment by eladkarako** [comment]
https://gist.github.com/eladkarako/ff2b36ff2f8e199d597bd903cc412bad#file-api-ms-win-files-for-windows-7-x86-x64-windows-8-1-x86-x64-md

TL;DR
<blockquote>
<em>This 'thread' is over 5 years old... <br/>
but it is still the 2<sup>nd</sup> result in my Google-search for searching <br/>
<code>API-MS-WIN-CORE-WINRT-STRING-L1-1-0.DLL</code> <br/>
<sub>(ignoring results from <code>dll-files.com</code>)</sub>
</em>
</blockquote>

Why?
Essentially everything that was built with MVS2015 and/or Win10SDK creates a dependency on something called <code>the Universal CRT</code> which are just few system-components.

Get files:
<ol>
<li>
You can get older OS to work (down to Windows Vista actually) by installing the related update from this website: https://support.microsoft.com/en-us/help/2999226/update-for-universal-c-runtime-in-windows.
</li>
<li>
You need to choose the version of OS to download, 
here is the direct-link for Windows 7 (x86/x64): <br/>
https://download.microsoft.com/download/4/F/E/4FE73868-5EDD-4B47-8B33-CE1BB7B2B16A/Windows6.1-KB2999226-x86.msu
https://download.microsoft.com/download/1/1/5/11565A9A-EA09-4F0A-A57E-520D5D138140/Windows6.1-KB2999226-x64.msu
</li>
<li>
Next, use "<code>Patch Extract v1.25 by Greg Linares (@Laughing_Mantis)</code>" <a href="https://gist.github.com/anonymous/d55f494982c0097111d3263cf7099c9d">https://gist.github.com/anonymous/d55f494982c0097111d3263cf7099c9d</a> to extract the files from the MSU files, neither 7zip nor UniExtract will help you with that. <br/>
You can see how I did it in my wrapping-script https://github.com/eladkarako/mods/blob/store/MSU-Extractor/start.cmd
</li>
<li>
You'll find your file under the <code>x86</code> or <code>x64</code> folder and the build you have (you might have two builds for example <code>u..rsalcrt-apifwd-win7_6.1.7601.18972</code> and <code>u..rsalcrt-apifwd-win7_6.1.7601.23175</code>), <br/>
if you have no specific version preferred (by a manifest file... md5 etc...) enter the latest one. <br/>
You'll found about 23 file with the following names (most likely): <br/>
<pre>
api-ms-win-core-file-l1-2-0.dll
api-ms-win-core-file-l2-1-0.dll
api-ms-win-core-localization-l1-2-0.dll
api-ms-win-core-processthreads-l1-1-1.dll
api-ms-win-core-synch-l1-2-0.dll
api-ms-win-core-timezone-l1-1-0.dll
api-ms-win-core-xstate-l2-1-0.dll
api-ms-win-crt-conio-l1-1-0.dll
api-ms-win-crt-convert-l1-1-0.dll
api-ms-win-crt-environment-l1-1-0.dll
api-ms-win-crt-filesystem-l1-1-0.dll
api-ms-win-crt-heap-l1-1-0.dll
api-ms-win-crt-locale-l1-1-0.dll
api-ms-win-crt-math-l1-1-0.dll
api-ms-win-crt-multibyte-l1-1-0.dll
api-ms-win-crt-private-l1-1-0.dll
api-ms-win-crt-process-l1-1-0.dll
api-ms-win-crt-runtime-l1-1-0.dll
api-ms-win-crt-stdio-l1-1-0.dll
api-ms-win-crt-string-l1-1-0.dll
api-ms-win-crt-time-l1-1-0.dll
api-ms-win-crt-utility-l1-1-0.dll
api-ms-win-eventing-provider-l1-1-0.dll
</pre>
</li>
<li>
You can get the Windows7 x86/x64 and Windows8.1 x86/x64 from this gist (MSU extraction done for you...) - just use 7zip to extract the archive and dig-for the content inside...
https://gist.github.com/eladkarako/ff2b36ff2f8e199d597bd903cc412bad#file-api-ms-win-files-for-windows-7-x86-x64-windows-8-1-x86-x64-md
</li>
</ol>
